pkg/election/election.go

```diff
@@ -158,6 +179,11 @@ func (e *ZooKeeperElection) waitForNodeDeletion(ctx context.Context, node string
 		return fmt.Errorf("node watch %s: %w", node, err)
 	}
 
+	baseWatch, err := e.zk.WatchChildren(e.basePath)
```

Comment:
Do you think it would make sense to rename the function? Just thinking it now may wake up on any node change (addition included) not just a specific node deletion.

Reply:
Yes indeed, will change.